projects/plugins/jetpack/modules/contact-form/admin.php

```diff
@@ -1128,3 +1113,263 @@ function grunion_feedback_admin_notice() {
+							<path d="M719.9 37l-4.8-8.9H694l-4.6 8.9h-7.1l19.5-36h5.6l19.8 36H719.9zM704.4 8l-7.8 15.1h15.9L704.4 8zM733 37V1h6.8v36H733zM781 37c-1.8 0-2.6-2.5-2.9-5.8l-0.2-3.7c-0.2-3.6-1.7-5.1-8.4-5.1h-12.8V37H750V1h19.6c10.8 0 15.7 4.3 15.7 9.9 0 3.9-2 7.7-9 9 7 0.5 8.5 3.7 8.6 7.9l0.1 3c0.1 2.5 0.5 4.3 2.2 6.1V37H781zM778.5 11.8c0-2.6-2.1-5.1-7.9-5.1h-13.8v10.8h14.4c5 0 7.3-2.4 7.3-5.2V11.8zM794.8 37V1h6.8v30.4h28.2V37H794.8zM836.7 37V1h6.8v36H836.7zM886.2 37l-23.4-24.1 -2.3-2.5V37h-6.8V1h6.5l22.7 24.1 2.3 2.6V1h6.8v36H886.2zM902.3 37V1H935v5.6h-26v9.2h20v5.5h-20v10.1h26V37H902.3z"></path>
+						</svg>
+					</div>
+				</div>
+			</div>
+		</div>
+		<?php
+		$opener_label        = esc_html( __( 'Export', 'jetpack' ) );
+		$export_modal_opener = "<a id='export-modal-opener' class='button button-primary' href='#TB_inline?&width=680&height=600&inlineId=feedback-export-modal'>{$opener_label}</a>";
+		?>
+		<script type="text/javascript">
+			jQuery( function( $ ) {
+				$( '#posts-filter #post-query-submit' ).after( <?php echo wp_json_encode( $export_modal_opener ); ?> );
+			} );
+		</script>
+		<?php
+	}
+
+	/**
+	 * Ajax handler for wp_ajax_grunion_export_to_gdrive.
+	 * Exports data to Google Drive, based on POST data.
+	 *
+	 * @see Grunion_Contact_Form_Plugin::get_feedback_entries_from_post
+	 */
+	public function export_to_gdrive() {
+		$post_data = wp_unslash( $_POST );
+		if ( empty( sanitize_text_field( $post_data[ $this->export_nonce_field ] ) ) || ! wp_verify_nonce( sanitize_text_field( $post_data[ $this->export_nonce_field ] ), 'feedback_export' ) ) {
+			wp_send_json_error(
+				__( 'You aren’t authorized to do that.', 'jetpack' ),
+				403
+			);
```

Comment:
Before we do that, I would recommend checking for permission, with `current_user_can( 'export' )`.

Reply:
I'm doing that check on the modal "print" function, but I guess I should be checking it here too 👍

Reply:
Yeah, definitely, since we cannot assume that the AJAX call will always be made via the modal.